### PR TITLE
feat(tooltip): support touch events

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #9273

## Summary

- support tap/touch events for tooltip
- cleanup
  - removed private vars for hoveredTooltip, clickedTooltip to simplify things
- add test

### CODE REVIEW

**Comment by driskull** [comment]
@jcfranco can we do this one for June release?

**Review by jcfranco — CHANGES_REQUESTED** [review]

File: packages/calcite-components/src/components/tooltip/TooltipManager.ts [hunk 1]
```diff
@@ -151,15 +163,15 @@ export default class TooltipManager {
   private addListeners(): void {
     window.addEventListener("keydown", this.keyDownHandler, { capture: true });
     window.addEventListener("pointermove", this.pointerMoveHandler, { capture: true });
-    window.addEventListener("click", this.clickHandler, { capture: true });
+    window.addEventListener("pointerdown", this.pointerDownHandler, { capture: true });
```
[the remark this answers is not in the document]
Comment (driskull) [reply packages/calcite-components/src/components/tooltip/TooltipManager.ts]: done! had to remove a test which is no longer appropriate.

**Comment by driskull** [comment]
@jcfranco ready for review again

**Comment by jcfranco** [comment]
It looks like there is a difference in behavior when clicking on an open tooltip:

**dev**

https://www.chromatic.com/component?appId=6266d45509d7eb004aa254fb&csfId=components-tooltip&buildNumber=5123&k=66760245ca218413c410f678-1200px-interactive-true&h=7&b=-1

**PR**

https://www.chromatic.com/component?appId=6266d45509d7eb004aa254fb&csfId=components-tooltip&buildNumber=5125&k=667617c3abd71b94f44673b8-1200px-interactive-true&h=3&b=-1

Not sure if it's related to the removed test.

**Comment by driskull** [comment]
@jcfranco I'm not seeing any difference there.

**Comment by jcfranco** [comment]
For posterity, reached out to Matt and a step I missed to include was to hover over the reference element to display the tooltip before clicking. We also spotted an issue where clicking the reference element to toggle the tooltip and then clicking on the tooltip would close for both dev and the PR.

**Comment by driskull** [comment]
> We also spotted an issue where clicking the reference element to toggle the tooltip and then clicking on the tooltip would close for both dev and the PR.

I think we'll have to ignore this one for now. The issue is that when you are focused on a reference element and then click on a tooltip, the onfocusout is fired which closes the open tooltip before the click can occur. 

I think we'll have to address this one as a followup if an issue arrises.

**Comment by driskull** [comment]
@jcfranco ready for review again.

File: packages/calcite-components/src/components/tooltip/tooltip.e2e.ts [hunk 2]
```diff
@@ -371,7 +371,7 @@ describe("calcite-tooltip", () => {
     expect(await tooltip.getProperty("open")).toBe(false);
   });
 
-  it("should not open tooltip when clicked", async () => {
+  it("should handle touch events", async () => {
```
Comment (jcfranco) [inline packages/calcite-components/src/components/tooltip/tooltip.e2e.ts:374]: This should be `mouse` or `pointer` events.

File: packages/calcite-components/src/components/tooltip/tooltip.e2e.ts [hunk 3]
```diff
@@ -388,7 +388,37 @@ describe("calcite-tooltip", () => {
 
     await page.evaluate(() => {
       const ref = document.getElementById("ref");
-      ref.click();
+      const event1 = new Event("click", {
```
Comment (jcfranco) [inline packages/calcite-components/src/components/tooltip/tooltip.e2e.ts:391]: Just curious, but is there a specific reason why `MouseEvent` or `PointerEvent` isn't being used here?

**Review by jcfranco — APPROVED** [review]
👈👈👈👈👈👈👈👈👈👈👈👈👈👈👈👈👈👈👈👈👈👈👈👈👈👈👈👈
👈🗯️👈👈👈🗯️👈👈🗯️🗯️👈👈🗯️🗯️🗯️👈👈🗯️🗯️🗯️👈🗯️🗯️🗯️🗯️👈🗯️👈
👈🗯️🗯️👈👈🗯️👈🗯️👈👈🗯️👈👈🗯️👈👈🗯️👈👈👈👈🗯️👈👈👈👈🗯️👈
👈🗯️👈🗯️👈🗯️👈🗯️👈👈🗯️👈👈🗯️👈👈🗯️👈👈👈👈🗯️🗯️🗯️👈👈🗯️👈
👈🗯️👈👈🗯️🗯️👈🗯️👈👈🗯️👈👈🗯️👈👈🗯️👈👈👈👈🗯️👈👈👈👈👈👈
👈🗯️👈👈👈🗯️👈👈🗯️🗯️👈👈🗯️🗯️🗯️👈👈🗯️🗯️🗯️👈🗯️🗯️🗯️🗯️👈🗯️👈
👈👈👈👈👈👈👈👈👈👈👈👈👈👈👈👈👈👈👈👈👈👈👈👈👈👈👈👈

File: packages/calcite-components/src/components/tooltip/tooltip.e2e.ts (hunk 3 above)
Comment (driskull) [reply packages/calcite-components/src/components/tooltip/tooltip.e2e.ts:391]: Nope, will update.

**Comment by jcfranco** [comment]
> I think we'll have to ignore this one for now. The issue is that when you are focused on a reference element and then click on a tooltip, the onfocusout is fired which closes the open tooltip before the click can occur.
> I think we'll have to address this one as a followup if an issue arrises.

@driskull Swap 'ignore' for 'handle separately' and we've got a deal! 😜 For tracking purposes, can you create a separate issue for this?

**Comment by driskull** [comment]
created https://github.com/Esri/calcite-design-system/issues/9827

**Pull request merged**